### PR TITLE
Add test and scenario for plugin TRIM

### PR DESCRIPTION
**Pull request by mateimicu**

### CODE REVIEW

File: argus/tests/cloud/windows/test_smoke.py [hunk 1]
```diff
@@ -215,3 +215,10 @@ class TestHeatUserdata(base.BaseTestCase):
     def test_heat_file_created(self):
         names = self._introspection.list_location('C:\\')
         self.assertIn('powershell_heat.txt', names)
+
+
+class TestTrimPlugin(base.BaseTestCase):
+
+    def test_trim_is_set(self):
+        value = self._introspection.get_trim_state()
+        self.assertEqual(True, value)
```
Comment (stefan-caraiman) [inline argus/tests/cloud/windows/test_smoke.py]: why not use `self.assertTrue(value)` instead?

File: ci/tests.py [hunk 2]
```diff
@@ -84,6 +84,9 @@ class ScenarioLongHostname(BaseWindowsScenario):
     recipe_type = recipe.CloudbaseinitLongHostname
     userdata = util.get_resource('windows/netbios_hostname')
 
+class ScenarioSimplePlugins(BaseWindowsScenario):
```
Comment (stefan-caraiman) [inline ci/tests.py]: A more appropriate name would be `ScenarioIndependentPlugins`

File: argus/introspection/cloud/windows.py [hunk 3]
```diff
@@ -317,6 +317,12 @@ def list_location(self, location):
             command, command_type=util.CMD)
         return list(filter(None, stdout.splitlines()))
 
+    def get_trim_state(self):
+        command = "fsutil.exe behavior query disabledeletenotify"
```
Comment (stefan-caraiman) [inline argus/introspection/cloud/windows.py]: A short docstring or note explaining this command would be useful

File: argus/recipes/cloud/windows.py [hunk 4]
```diff
@@ -538,6 +538,32 @@ def prepare_cbinit_config(self, service_type):
             name='netbios_host_name_compatibility', value='False')
 
 
+class CloudbaseinitEnableTrim(CloudbaseinitRecipe):
+    """Recipe for testing TRIM manipulation."""
```
Comment (ader1990) [inline argus/recipes/cloud/windows.py]: trim configuration*

File: argus/recipes/cloud/windows.py [hunk 5]
```diff
@@ -538,6 +538,32 @@ def prepare_cbinit_config(self, service_type):
...
+            name='trim_enabled', value='True')
+
+        self._cbinit_conf.set_conf_value(
+            name="plugins",
+            value="cloudbaseinit.plugins.common.trim"
+                  ".TrimConfigPlugin")
+
+
+class CloudbaseinitIndependentPlugins(CloudbaseinitEnableTrim):
+    """Recipe for simple plugins."""
```
Comment (ader1990) [inline argus/recipes/cloud/windows.py]: independent plugins*

**Review by ader1990 — CHANGES_REQUESTED** [review]

File: argus/introspection/cloud/windows.py [hunk 6]
```diff
@@ -317,6 +317,15 @@ def list_location(self, location):
...
         return list(filter(None, stdout.splitlines()))
 
+    def get_trim_state(self):
+        # Query the curent stat of DisableDeleteNotify
+        # 1 - DeleteNotify is disabled
+        # 0 - DeleteNotify is enabled
+        command = "fsutil.exe behavior query disabledeletenotify"
+        stdout = self.remote_client.run_command_verbose(
+            command, command_type=util.CMD)
+        return "= 0" in stdout
```
Comment (ader1990) [inline argus/introspection/cloud/windows.py]: check if DisableDeleteNotify = 0

**Review by ader1990 — CHANGES_REQUESTED** [review]

**Review by ader1990 — APPROVED** [review]

**Pull request merged**